### PR TITLE
Updated signup form on homepage's take action section

### PR DESCRIPTION
**Pull request by mmmavis**
Use the new React components to render the signup form on homepage's "take action" section.

Link to sample test page: https://foundation-s-11656-take-zqfpgh.herokuapp.com/en/
Related PRs/issues: #11656

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/take_action.html [hunk 1]
```diff
@@ -28,12 +28,14 @@ <h2 class="take-action-title tw-mb-12">{{ page.take_action_title }}</h2>
...
-                <div
-                    class="join-us"
+                <div class="newsletter-signup-module"
+                    data-module-type="default"
+                    data-form-style="home-body"
                     data-form-position="body"
+                    data-button-text="{% trans "Sign up" context "Submit button for newsletter signup form" %}"
                     data-cta-header="{% trans "Help shape the future of the web for the public good." %}"
-                    data-cta-description="<p class='mb-3'>{% trans "Join our <b>Mozilla News</b> email list to get action alerts & internet tips right to your inbox." %}</p>"
-                    data-newsletter="{{ cta.newsletter }}">
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/take_action.html:36]: `cta` is not available on the homepage so this `data-newsletter` attribute basically does nothing

File: network-api/networkapi/templates/fragments/footer.html [hunk 2]
```diff
@@ -32,7 +32,7 @@
                                  data-form-position="footer"
                                  data-button-text="{% trans "Sign up" context "Submit button for newsletter signup form" %}"
                                  data-cta-header="{% trans "Help shape the future of the web for the public good." %}"
-                                 data-cta-description="{% trans "<p>Join our <b>Mozilla News</b> email list to get action alerts & internet tips right to your inbox.</p>" %}"
+                                 data-cta-description="<p>{% trans "Join our <b>Mozilla News</b> email list to get action alerts & internet tips right to your inbox." %}</p>"
```
Comment (mmmavis) [inline network-api/networkapi/templates/fragments/footer.html:35]: Changed this to re-use the string from `take_action.html` signup form (which has already been translated)

File: network-api/networkapi/templates/fragments/primary_nav.html [hunk 3]
```diff
@@ -88,7 +88,7 @@
                      data-form-position="header"
                      data-button-text="{% trans "Sign up" context "Submit button for newsletter signup form" %}"
                      data-cta-header="{% trans "Help shape the future of the web for the public good." %}"
-                     data-cta-description="{% trans "<p>Join our <b>Mozilla News</b> email list to get action alerts & internet tips right to your inbox.</p>" %}"
+                     data-cta-description="<p>{% trans "Join our <b>Mozilla News</b> email list to get action alerts & internet tips right to your inbox." %}</p>"
```
Comment (mmmavis) [inline network-api/networkapi/templates/fragments/primary_nav.html:91]: Changed this to re-use the string from `take_action.html` signup form (which has already been translated)

**Review by danielfmiranda — APPROVED** [review]
Reviewed and tested. Everything looks great to me! 

Thanks @mmmavis 🙌

**Pull request merged**